src/main/java/org/mule/module/xml/internal/xpath/XPathEvaluator.java

```diff
@@ -115,15 +118,23 @@ public XPathEvaluator(String expression, XPathFactory xpathFactory, Collection<N
     for (int i = 0; i < size; i++) {
       StringWriter sw = new StringWriter();
       toString.transform(new DOMSource(nodeList.item(i)), new StreamResult(sw));
-      strings.add(sw.toString());
+      // Since Saxon 9.9.1-1, the "indented" serializer options seems to add a new-line character
+      // after each DOM Node. Could not found a property to avoid this behaviour.
+      strings.add(cleanTrailingNewlineIfNecessary(sw.toString()));
```

Comment:
why is it you need this here but you didn't in the operation?

Reply:
The operation sets wether the method ```cleanTrailingNewlineIfNecessary``` should clean the newlines with the method ```keepingTrailingNewlines(addTrailingNewlines)```. I could clean the new lines directly from the operation, but it seemed easier in the Evaluator since it already iterates over all responses.